src/libaktualizr/storage/sqlstorage.h

```diff
@@ -59,6 +59,10 @@ class SQLStorage : public INvStorage {
                               const std::string& current_hash) override;
   std::string loadInstalledVersions(std::vector<Uptane::Target>* installed_versions) override;
   void clearInstalledVersions() override;
+  void storeInstallationResult(const std::string& installation_result) override;
```

Comment:
How about storing and getting `data::OperationResult`, so that the json conversions (in storage implementations) are guaranteed to be compatible with each others instead of relying on api user code?

Reply:
So you mean store each of the constituent elements of `data::OperationResult` instead of the serialized json? Not a bad idea. I'm not sure I understand what you mean about compatibility and api user code, though.

Reply:
Thinking about this more, I like the idea of storing actual types instead of a json string, but it gets annoying for the old filesystem storage. I'm probably going to have to store three separate files with each of the members of `data::OperationResult`. I would love it if we could get rid of filesystem storage!

Reply:
What I meant is that passing `std::string` around is close to being untyped, users of the api need to construct the correct json object manually and nothing prevents them to go out of sync.

If this is impractical, nevermind. About `FSStorage`, I also think it's worth drafting a deprecation plan and would gladly start it.

Reply:
I have a plan for the `data::OperationResult` storage. I think we are on the same page now.

Deprecating `FSStorage` might be impractical, but we could at least move to a state where we do not allow it as a valid config option anymore. We probably still have to maintain the existing logic to be able to convert from `FSStorage` to `SQLStorage`, though. But at least we could stop implementing new features for it. The first step of the process is converting the default implicit and HSM provisioning configs to use `SQLStorage` as well as any tests that aren't specifically designed to test `FSStorage`.